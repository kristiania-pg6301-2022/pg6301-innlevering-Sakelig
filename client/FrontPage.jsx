import { useLoader } from "./useLoader";
import { fetchJSON } from "./http";
import { Link } from "react-router-dom";
import * as React from "react";

export function FrontPage({ quizApi }) {
  const {
    data: score,
    loading,
    error,
    reload,
  } = useLoader(async () => await quizApi.getScore());

  if (error) {
    return <div>Something went wrong D:</div>;
  }

  return (
    <div>
      <h1>Quiz !</h1>
      {loading && <div>Loading ...</div>}
      {score && (
        <h3>
          You have answered {score.correct} of {score.answers} correctly so far!
        </h3>
      )}

      <ul>
        <li>
          <Link to={"/question"}>ny quiz</Link>
        </li>
      </ul>
    </div>
  );
}
