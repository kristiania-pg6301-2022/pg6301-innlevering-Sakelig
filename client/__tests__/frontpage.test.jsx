import * as React from "react";
import * as ReactDOM from "react-dom";
import { FrontPage } from "../FrontPage";
import { MemoryRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";

describe("Frontpage", () => {
  it("show unloaded frontpage", () => {
    const getScore = () => new Promise((resolve) => {});

    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FrontPage quizApi={{ getScore }} />
      </MemoryRouter>,
      element
    );

    expect(element.innerHTML).toMatchSnapshot();
  });

  it("show loaded frontpage", async () => {
    const getScore = () => ({
      answers: 3,
      correct: 2,
    });

    const element = document.createElement("div");

    await act(async () => {
      await ReactDOM.render(
        <MemoryRouter>
          <FrontPage quizApi={{ getScore }} />
        </MemoryRouter>,
        element
      );
    });
    expect(element.innerHTML).toMatchSnapshot();
    expect(element.querySelector("h3").textContent).toContain("2");
  });

  it("show get error frontpage", async () => {
    const getScore = () => {
      throw new Error("Failed to load");
    };

    const element = document.createElement("div");

    await act(async () => {
      await ReactDOM.render(
        <MemoryRouter>
          <FrontPage quizApi={{ getScore }} />
        </MemoryRouter>,
        element
      );
    });
    expect(element.querySelector("div").textContent).toContain(
      "Something went wrong D:"
    );
  });
});
