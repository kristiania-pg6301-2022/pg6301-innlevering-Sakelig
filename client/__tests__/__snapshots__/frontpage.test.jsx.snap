// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontpage show loaded frontpage 1`] = `"<div><h1>Quiz !</h1><h3>You have answered 2 of 3 correctly so far!</h3><ul><li><a href=\\"/question\\">ny quiz</a></li></ul></div>"`;

exports[`Frontpage show unloaded frontpage 1`] = `"<div><h1>Quiz !</h1><div>Loading ...</div><ul><li><a href=\\"/question\\">ny quiz</a></li></ul></div>"`;
